src/net/sourceforge/kolmafia/shop/ShopRowDatabase.java

```diff
@@ -32,52 +32,57 @@ enum Mode {
   // If we eventually need this internally, set to Mode.READ
   // If you want to regenerate the data file, set to Mode.BUILD
   // recompile, log in, and "test write-shoprows"
-  public static Mode mode = Mode.NONE;
+  public static Mode mode = Mode.READ;
```

Comment:
if deliberate please change the comment above

Reply:
It was deliberate, because I wanted access to ShopRowData elsewhere, without having to force a load - except that might only be in TestCommand. Unless further refactoring requires this in production code, no need to read the file.

I have specific refactorings intended for RequestLogger and ResponseTextParser and NPCPurchaseRequest; the end result may be that coinmasters requests won't need to provide certain methods, since ShopRequest will provide them.

If I do that, ShopRequest will need this to look up the ingredients that were consumed.

Let me get some coffee and think about this; if I don't need READ now, I'll defer until I do.

Reply:
OK. ShopRequest.registerRequest does this: ``` ShopRow shopRow = ShopRowDatabase.getShopRow(row);```

We don't call that method yet - but we will soon.

I'll change the mode back to NONE, and when I actually start needing the data, I'll set it to READ and change the comment as suggested.

Reply:
Fixed.